cmd/api/src/daemons/datapipe/datapipe.go

```diff
@@ -182,6 +206,25 @@ func (s *Daemon) Start(ctx context.Context) {
 	}
 }
 
+func (s *Daemon) deleteData() {
+	defer func() {
+		s.status.Update(model.DatapipeStatusIdle, false)
+		s.setDeletionRequested(false)
+		s.setAnalysisRequested(true)
+	}()
+	defer log.Measure(log.LevelInfo, "Purge Graph Data Completed")()
+	s.status.Update(model.DatapipeStatusPurging, false)
```

Comment:
should this be setting the purging state to true?

Reply:
This method has a confusing signature if you're not looking at it in an IDE. The bool flag is to say whether to update analysis time, which we don't want to do because this isn't an analysis operation. Regardless of what you set that flag to, the status you give it will become the current status for the datapipe.